webapp/src/services/api.ts

```diff
@@ -18,8 +18,9 @@ const responseToData =
     try {
       const response = await responsePromise(arg);
       return { data: await response.json() };
-    } catch {
-      return { error: { message } };
+    } catch (err) {
+      const error = err as Error;
+      return { error: { message: message + error.message } };
```

Comment:
Also, the implicit conversion of ar error to a string is the same as accessing the `.message` property, so no need to assert the type and everything here:
```suggestion
    } catch (error) {
      return { error: { message: `${message}\n${error}` } };
```